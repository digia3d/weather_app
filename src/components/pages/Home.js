import { React, useEffect } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { useDispatch, useSelector } from 'react-redux';

import { getWeather } from '../redux/Weathers';
import SinglCityWeather from './City';

function Home() {
  const weathers = useSelector((state) => state.weathers);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getWeather());
  }, [dispatch]);
  return (
    <>
      <div className="info">Forecast for five days</div>
      <div className="weather-card">
        {weathers.map((weather, index) => (
          <SinglCityWeather
            key={uuidv4()}
            page={index + 1}
            id={weather.weather.id}
            weather={weather.weather[0].main}
            icon={weather.weather[0].icon}
            temp={weather.main.temp}
            feels_like={weather.main.feels_like}
            temp_min={weather.main.temp_min}
            temp_max={weather.main.temp_max}
            pressure={weather.main.pressure}
            humidity={weather.main.humidity}
            speed={weather.wind.speed}
            deg={weather.wind.deg}
            gust={weather.wind.gust}
            all={weather.clouds.all}
            date={weather.dt_txt}
            country=""
            sunrise=""
            sunset="{weather.sunset}"
            timezone="{weather.timezone}"
            name={weather.weather[0].description}
          />
        ))}
      </div>
    </>
  );
}
export default Home;
