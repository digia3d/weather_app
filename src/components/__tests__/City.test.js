import React from "react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Home from '../pages/Home';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';
import { Provider } from "react-redux";

describe(
  'Home', () => {
    it(
      'should render the component', () => {
        const tree = renderer.create(
          <React.StrictMode>
            <Provider store={store}>
              <BrowserRouter>
                <Home />
              </BrowserRouter>
            </Provider>
          </React.StrictMode>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
      }
    )
  }
)
