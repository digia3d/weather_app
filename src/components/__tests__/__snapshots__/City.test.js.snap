// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the component 1`] = `
Array [
  <div
    className="info"
  >
    Forecast for five days
  </div>,
  <div
    className="weather-card"
  />,
]
`;
